tests/test_SequenceManager.cc

```diff
@@ -82,6 +82,49 @@ TEST_CASE("SequenceManager: Move constructor", "[SequenceManager]")
     REQUIRE(s.get_path() == "unit_test_files");
 }
 
+TEST_CASE("SequenceManager: copy_sequence()", "[SequenceManager]")
+{
+    const char* dir = "unit_test_files/copy_sequence_test";
+
+    if (std::filesystem::exists(dir))
+        std::filesystem::remove_all(dir);
+
+    std::filesystem::create_directories(dir);
+    SequenceManager manager{ dir };
+
+    // Create a sequence and store it
+    Sequence seq{ "First sequence", SequenceName{ "first" } };
+    Step step{ Step::type_action };
+    step.set_script("answer = 42");
+    seq.push_back(step);
+    manager.store_sequence(seq);
+
+    // Make a copy
+    Sequence copy = manager.copy_sequence(seq.get_unique_id(), SequenceName("copy"));
+
+    // Compare original and copy
+    REQUIRE(copy.get_label() == "First sequence");
+    REQUIRE(copy.get_name() == SequenceName{ "copy" });
+    REQUIRE(copy.get_unique_id() != seq.get_unique_id());
```

Comment:
Its good to check the dissimilarity of the id but for me it is a **weak test** because it checks on two random 64bit integers. The possibility that they are the same is very rare but can exists. When it fails we need to review the algorithm about unique ids (it smells a little bit ;-).

Reply:
Well, we could have forgotten to generate a new unique ID for the copied sequence...